rados/snapshot.go

```diff
@@ -70,3 +75,77 @@ func (snapshot *Snapshot) Remove() error {
 	ret := C.rados_ioctx_snap_remove(snapshot.ioctx.ioctx, cSnapName)
 	return getError(ret)
 }
+
+// SnapID represents the ID of a rados snapshot.
+type SnapID C.rados_snap_t
+
+// LookupSnapshot gets the ID of a pool snapshot.
+//
+// Implements:
+//  int rados_ioctx_snap_lookup(rados_ioctx_t io, const char *name, rados_snap_t *id)
+func (snapshot *Snapshot) LookupSnapshot() (SnapID, error) {
+	var snapID SnapID
+
+	if err := snapshot.validate(); err != nil {
+		return snapID, err
+	}
+
+	cSnapName := C.CString(snapshot.name)
+	defer C.free(unsafe.Pointer(cSnapName))
+
+	ret := C.rados_ioctx_snap_lookup(
+		snapshot.ioctx.ioctx,
+		cSnapName,
+		(*C.rados_snap_t)(&snapID))
+	return snapID, getError(ret)
+}
+
+// GetSnapName fetches the name of a pool snapshot with the given snapshot ID.
+//
+// Implements:
+//	int rados_ioctx_snap_get_name(rados_ioctx_t io, rados_snap_t id, char *name, int maxlen)
+func (ioctx *IOContext) GetSnapName(snapID SnapID) (string, error) {
+	if ioctx.ioctx == nil {
+		return "", ErrNoIOContext
+	}
+
+	var (
+		buf []byte
+		err error
+	)
+	// range from 1k to 64KiB
+	retry.WithSizes(1024, 1<<16, func(len int) retry.Hint {
+		cLen := C.int(len)
+		buf = make([]byte, cLen)
+		ret := C.rados_ioctx_snap_get_name(
+			ioctx.ioctx,
+			(C.rados_snap_t)(snapID),
+			(*C.char)(unsafe.Pointer(&buf[0])),
+			cLen)
+		err = getError(ret)
+		return retry.Size(int(cLen)).If(err == errRange)
+	})
+
+	if err != nil {
+		return "", err
+	}
+	return C.GoString((*C.char)(unsafe.Pointer(&buf[0]))), nil
+}
+
+// GetSnapStamp returns the time of the pool snapshot creation.
+//
+// Implements:
+//  int rados_ioctx_snap_get_stamp(rados_ioctx_t io, rados_snap_t id, time_t *t)
+func (ioctx *IOContext) GetSnapStamp(snapID SnapID) (time.Time, error) {
```

Comment:
This could be a method of Snapshot too, maybe call it Snapshot.GetStamp() and fetch the SnapID if the Snapshot object does not have it set yet

Reply:
Sure, only think is that Lookup() has to be called everytime before calling this. Is that ok?

Reply:
The Snapshot type can have an ID, if it is not set then Lookup() can be called internally, and skipped the next time.